robowflex_doc/doc/markdown/rviz.md

```diff
@@ -0,0 +1,65 @@
+# Live Visualization with RViz {#rviz}
+
+Robowflex offers a helper class, robowflex::IO::RVIZHelper, which makes it easy to display information in RViz from scripts that use Robowflex.
+This document gives a brief tutorial of how you might use RVIZHelper in your scripts.
+You can also look at [ur5_visualization.cpp](ur5__visualization_8cpp_source.html) which demonstrates RVIZHelper.
+
+## Using RVIZHelper in a Script
+
+You first need to create your robot and setup ROS:
+```cpp
+// Startup ROS
+ROS ros(argc, argv);
+
+// Create the default UR5 robot.
+auto ur5 = std::make_shared<UR5Robot>();
+ur5->initialize();
+```
+
+With your robot, we can create the helper:
+```cpp
+IO::RVIZHelper rviz(ur5);
+```
+
+This republishes the robot's information under an easily typable name on the parameter server.
+By default, the name is `/robowflex/robot_description` and `/robowflex/robot_description_semantic`.
+However, you can change this if you are visualizing multiple robots:
+```cpp
+IO::RVIZHelper rviz(ur5, "ur5");
+```
+
+This will publish `/ur5/robot_description` and `/ur5/robot_description_semantic`.
+
+You should probably insert a pause in your program at this point, as you need to setup RViz to look at these topics.
+
+## Setting up RViz
+
+Robowflex namespaces things to avoid conflicting with other ROS happenings, but this means RViz needs a little configuration.
+By default, the helper publishes scenes under `moveit_msgs::PlanningScene`, trajectories under `moveit_msgs::DisplayTrajectory`, and markers under `visualization_msgs::MarkerArray`.
+You can display these in RViz with `PlanningScene`, `Trajectory`, and `MarkerArray` widgets.
+`PlanningScene` and `Trajectory` need the robot description topic set to whatever you are publishing the robot under (see above).
+They also need the topic adjusted to the namespace as well.
+
+Here is an example with the default configuration:
+<img style="width:50%;max-width:720px;" src="https://s3.amazonaws.com/zk-bucket/robowflex/rviz.png" />
```

Comment:
I put them on my amazon bucket to save space, but we can definitely do another pull after this to 1) update the documentation to what's currently on master and 2) bring images into the repo, rather than be externally hosted.